slides/05_First_Components.html

```diff
@@ -432,35 +428,137 @@ <h3>Solution continued</h3>
-                        }
+                            );
+                        }</script></code></pre>
+          </section>
+          <section data-auto-animate>
+            <!-- prettier-ignore -->
+            <pre data-id="code"><code class="hljs javascript" data-trim data-line-numbers="6"><script type="text/template">
+                        export default function List() {
+                            const items = ['Apple', 'Orange', 'Mango'];
+                            return (
+                                <ul>
+                                    {items.map(
+                                        item => <li key={item}>{item}</li>
+                                    )}
+                                </ul>
+                            );
+                        }</script></code></pre>
 
-                        export default List;</script></code></pre>
+            <p>
+              The JSX element returned from <span class="code">map</span> must
+              have a <span class="warn">key</span> prop.
+            </p>
+
+            <p class="fragment">
+              This is a special prop used by React internally to distinguish the
+              elements. It must be
+              <span class="warn">unique within the list</span> (e.g. an entity
+              ID).
+            </p>
           </section>
+        </section>
+        <section>
+          <section><h2>The children prop</h2></section>
+          <section>
+            <p>Like HTML elements, React components can be nested:</p>
+
+            <pre>
+            <code class="hljs jsx" data-trim>
+                <script type="text/template">
+                export default function App() {
+                  return (
+                    <ContentContainer title="Profile">
+                      <ProfilePicture />
+                      <Bio />
+                      <ScrollableSection>
+                        <ProjectList />
+                      </ScrollableSection>
+                    </ContentContainer>
+                  );
+                }
+                </script>
+            </code>
+        </pre>
+          </section>
+          <section data-auto-animate>
+            <p>
+              The parent component receives the child components through the
+              <span class="warn">children</span> prop.
+            </p>
+
+            <pre>
+            <code class="hljs jsx" data-trim data-line-numbers="1,5,14,16" data-id="code">
+                <script type="text/template">
+                import { PropsWithChildren } from "react"
+
+                function ContentContainer({
+                  title,
+                  children,
+                }: ContentContainerProps) {
+                  return (
+                    <div className="content-container">
+                      …
+                    </div>
+                  );
+                }
+
+                type ContentContainerProps = PropsWithChildren<{
```

Comment:
For me a bit strange order, I do normally have the Props defined before the component.

Reply:
I like that practice as you describe it, I should stick to that too.